jina/executors/__init__.py

```diff
@@ -198,6 +198,24 @@ def __call__(self, req_endpoint: str, **kwargs):
                 self, **kwargs
             )  # unbound method, self is required
 
+    async def __acall__(self, req_endpoint: str, **kwargs):
+        """
+        # noqa: DAR101
+        # noqa: DAR102
+        # noqa: DAR201
+        """
+        if req_endpoint in self.requests:
+            return await self.__acall_endpoint__(req_endpoint, **kwargs)
+        elif __default_endpoint__ in self.requests:
+            return await self.__acall_endpoint__(__default_endpoint__, **kwargs)
+
+    async def __acall_endpoint__(self, req_endpoint, **kwargs):
+        func = self.requests[req_endpoint]
+        if iscoroutinefunction(func):
+            return await (await func(self, **kwargs))  # 2 awaits ugly
+        else:
+            return await run_in_threadpool(func, self, **kwargs)
```

Comment:
why is this needed, shouldnt it simply fail? Running in thread is undesired

Reply:
Here we just simply do asyncio.run()

Reply:
The event loop is already running inside the Executor. So we cannot do `asyncio.run` or `loop.run_until_complete`. Anyway, doing that will block the event loop. The non async defs are `run_in_executor` to keep the event loop unblocked.

Reply:
> why is the speedup not felt?

?? Before the PR, sending 5 requests with used to take 5 secs. Now it takes one (both normal def and async def). The event loops remains unblocked throughout.

Reply:
ok not understood. But in the non async def it takes 1 sec because u run it on a thread right?